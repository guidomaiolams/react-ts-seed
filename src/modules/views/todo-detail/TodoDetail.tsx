import React, { memo, useEffect } from 'react';

import { TodoModel } from '../../models';

export interface ITodoDetailProps {
  todo: TodoModel.ITodo;
  fetchTodoDetail: () => void;
}

const TodoDetail = ({ todo, fetchTodoDetail }: ITodoDetailProps) => {
  useEffect(() => {
    fetchTodoDetail();
  }, []); // eslint-disable-line

  return (
    <div className="container" key="TodoDetail">
      <h1>
        {todo.id} {todo.name}
      </h1>
      <p>Description: {todo.description}</p>
    </div>
  );
};

export default memo(TodoDetail);
