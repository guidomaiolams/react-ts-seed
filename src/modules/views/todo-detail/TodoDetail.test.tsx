import React from 'react';
import { mount } from 'enzyme';
import { create } from 'react-test-renderer';

import { getTodo_1 } from '../../../test/entities';
import TodoDetail from './TodoDetail';

describe('TodoList Component', () => {
  let Component;
  let props;

  beforeEach(() => {
    global.console.error = () => {/** */};
    props = {
      todo: { [getTodo_1().id]: getTodo_1() },
      fetchTodoDetail: jest.fn()
    };
    Component = mount(<TodoDetail {...props} />);
  });

  it('should render with default props', () => {
    expect(create(Component).toJSON()).toMatchSnapshot();
  });
});
