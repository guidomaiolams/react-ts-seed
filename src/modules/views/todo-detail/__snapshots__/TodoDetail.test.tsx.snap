// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList Component should render with default props 1`] = `
<div
  className="container"
>
  <h1>
     
  </h1>
  <p>
    Description: 
  </p>
</div>
`;
